channeldb/control_tower.go

```diff
@@ -64,18 +71,39 @@ type ControlTower interface {
 
 	// FetchInFlightPayments returns all payments with status InFlight.
 	FetchInFlightPayments() ([]*InFlightPayment, error)
+
+	// SubscribePayment subscribes to updates for the payment with the given
+	// hash. It returns a boolean indicating whether the payment is still in
+	// flight and a channel that provides the final outcome of the payment.
+	SubscribePayment(paymentHash lntypes.Hash) (bool, chan PaymentEvent,
```

Comment:
How are subscriptions cancelled? To me, this feel like something that should be done on a higher level (in the router for example by intercepting the calls to the interface). Nowhere else do we have subscription/event logic _within_ the primary database package.

Reply:
Subscriptions don't need to be canceled. There is only ever a single notification and the channel is buffered.

Reply:
You are absolutely right about the abstraction. Created a split between db and subscriber handling.